### PR TITLE
ROX-16208: Sensor UpdateImage integration test

### PR DESCRIPTION
**Pull request by fredrb**
## Description

This PR provides a test that captures the issue found in the new re-sync pipeline related to `UpdateImages` not being properly handled when scanner updates image components or wasn't up when a deployment is initially processed. 

The issue is fixed by #5365, however, we didn't have any automated tests failed before. This new test will make sure the issue is fixed and also serve as regression in case the code changes.

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~~Evaluated and added CHANGELOG entry if required~~
- [ ] ~~Determined and documented upgrade steps~~
- [ ] ~~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

## Testing Performed

Just the added tests

### CODE REVIEW

File: sensor/tests/resource/imagescan/imagescan_test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+						},
+					},
+				},
+			})
+
+			tc.GetFakeCentral().StubMessage(&central.MsgToSensor{
+				Msg: &central.MsgToSensor_ReprocessDeployments{
+					ReprocessDeployments: &central.ReprocessDeployments{},
+				},
+			})
```
Comment (lvalerom) [inline sensor/tests/resource/imagescan/imagescan_test.go:125]: If we decide to trigger affected deployments on `UpdatedImage`, then the `ReprocessDeployments` shouldn't be needed here. I think, unless there's a guarantee that after `UpdatedImage` we are going to receive `ReprocessDeployments`, we should trigger affected deployments on `UpdatedImage`, thus I think the tests should test this without sending a `ReprocessDeployments`.

**Review by lvalerom — CHANGES_REQUESTED** [review]
Looks good. I'm just blocking because I think we should agree on sending or not the `ReprocessDeployments` message before merging.

**Comment by fredrb** [comment]
> Looks good. I'm just blocking because I think we should agree on sending or not the ReprocessDeployments message before merging.

I don't agree with this approach. The current behavior (even before your PR) is that Central _has_ to send `ReprocessDeployment` for the deployments to be reprocessed -- even with re-sync enabled. So this test is just validating a constraint we have in the system. I prefer to have this test running in CI and PRs so we can also make sure it passes with your change once is merged.

If we make the change that `UpdateImage` is sufficient to reprocess deployments, we can update the test to reflect that.

File: sensor/tests/resource/imagescan/imagescan_test.go (hunk 1 above)
Comment (fredrb) [reply sensor/tests/resource/imagescan/imagescan_test.go:125]: See comment below

**Review by lvalerom — APPROVED** [review]

**Pull request merged**